sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVOptions.scala

```diff
@@ -212,6 +212,11 @@ class CSVOptions(
     settings.setEmptyValue(emptyValueInRead)
     settings.setMaxCharsPerColumn(maxCharsPerColumn)
     settings.setUnescapedQuoteHandling(UnescapedQuoteHandling.STOP_AT_DELIMITER)
+
+    if (multiLine) {
+      settings.setLineSeparatorDetectionEnabled(true)
```

Comment:
Would be simple just `settings.setLineSeparatorDetectionEnabled(multiLine)` or `settings.setLineSeparatorDetectionEnabled(multiLine == true)`?